LogfileIterator/PolySyncLogfile.cpp

```diff
@@ -0,0 +1,501 @@
+{
+    checkDTC( psync_logfile_enable_output_queue
+              ( _node, enableReplayMessageQueue ),
+                "Failed to enable LogFile Replay Message Queue." );
+}
+
+
+void Logfile::getThreadStatus()
+{
+    checkDTC( psync_logfile_get_thread_status( _node ),
+                "Failed to get/check LogFile thread status." );
+}
+
+
+void Logfile::setReplayClockScaler(const double scale)
+{
+    checkDTC( psync_logfile_set_replay_clock_scaler( _node, scale ),
+                "Failed to set LogFile Replay Clock Scaler." );
+}
+
+
+double * const Logfile::getReplayClockScaler()
+{
+    double * scale = nullptr;
+
+    checkDTC( psync_logfile_get_replay_clock_scaler( _node, scale ),
+                "Failed to get LogFile Replay Clock Scaler." );
+
+    return scale;
+}
+
+
+void Logfile::setMessageTypeFilters(
+    ps_msg_type * const writerDisabledTypes,
+    const unsigned long writerDisabledTypesLen,
+    ps_msg_type * const readerDisabledTypes,
+    const unsigned long readerDisabledTypesLen)
+{
+    checkDTC( psync_logfile_set_message_type_filters
+              ( _node, writerDisabledTypes, writerDisabledTypesLen,
+                readerDisabledTypes, readerDisabledTypesLen),
+                "Failed to set LogFile Message Type Filters." );
+}
+
+
+GAsyncQueue * Logfile::getReplayMessageQueue()
+{
+    GAsyncQueue * queuePtr = nullptr;
+
+    checkDTC( psync_logfile_get_replay_msg_queue( _node, &queuePtr ),
+                "Failed to get LogFile Replay Message Queue." );
+
+    return queuePtr;
+}
+
+
+ps_logfile_mode_kind Logfile::getMode()
+{
+    ps_logfile_mode_kind modeKind = LOGFILE_MODE_INVALID;
+
+    checkDTC( psync_logfile_get_mode( _node, &modeKind ),
+                "Failed to get LogFile mode." );
+
+    return modeKind;
+}
+
+
+ps_logfile_state_kind Logfile::getState()
+{
+    ps_logfile_state_kind stateKind = LOGFILE_STATE_INVALID;
+
+    checkDTC( psync_logfile_get_state( _node, &stateKind ),
+                "Failed to get LogFile state." );
+
+    return stateKind;
+}
+
+
+std::pair<ps_guid, ps_identifier> Logfile::getReplayClockOwner()
+{
+    std::pair<ps_guid, ps_identifier>getReplayClockOwnerPair;
+
+    ps_guid ownerGUID;
+
+    ps_identifier ownerPID;
+
+        checkDTC( psync_logfile_get_replay_clock_owner
+                  ( _node, &ownerGUID, &ownerPID ),
+                    "Failed to get LogFile Replay Clock Owner." );
+
+    return { ownerGUID, ownerPID };
+}
+
+
+bool Logfile::readerGetEofStatus()
+{
+    uint eofReached = 0;
+
+    checkDTC( psync_logfile_reader_get_eof_status
+              ( _node, &eofReached ),
+                "Failed to get LogFile EOF status." );
+
+    if ( eofReached )
+    {
+        return true;
+    }
+    return false;
+}
+
+
+void Logfile::readerResetToBeginning()
+{
+    checkDTC( psync_logfile_reader_reset_to_beginning( _node ),
+                "Failed to reset LogFile Reader to beginning of file." );
+}
```

Comment:
This function wraps the C Logfile API which returns DTC's. For the CPP Logfile Example purposes, this function is never invoked.